### ISSUE TITLE
[not spam !]  ratelimit error specific (Chinese query)

### ISSUE THREAD

**Issue by yichaosun** [deedy5/duckduckgo_search#262]
meet rate limit when the q is Chinese! when I replace the q with English, it worked!!

I have notice the ratelimit question , you closed, with lots of test, I found that Chinese is not work!
but when I use the curl（with the http url,  in ratelimit error msg） directly to DuckDuckGo, it works.

`duckduckgo_search  6.3.4`

```
from duckduckgo_search import DDGS


def main(q: str) -> dict:

  ddgs = DDGS()
  results = ddgs.text(q, region="cn-zh",max_results=5)
  return {
    "results": results,
    "total": len(results)
  }


print(main("中文"))
```

**Comment by yichaosun** [comment]
> Try querying in Chinese in your browser. Do you see captcha?

in browser, it worked
<img width="811" alt="image" src="https://github.com/user-attachments/assets/41fdb4fb-c2cc-4793-953d-ffd58153a8b0">


using  duckduckgo_search with chinese, ratelimit
<img width="1180" alt="image" src="https://github.com/user-attachments/assets/8b6ea6a5-4f0a-402a-8d55-75f9b5d62c92">

according to the error msg:
```
  File "/Users/sunyichao/Documents/Programer/Github/matrix/playwright/lib/python3.13/site-packages/duckduckgo_search/duckduckgo_search.py", line 132, in _get_url
    raise RatelimitException(f"{resp.url} {resp.status_code} Ratelimit")
duckduckgo_search.exceptions.RatelimitException: https://duckduckgo.com/?q=%E4%B8%AD%E6%96%87 202 Ratelimit
```
we do this, curl is worked
```
playwright➜  duckduckgo git:(main) ✗ curl https://duckduckgo.com/\?q\=%E4%B8%AD%E6%96%87
<!DOCTYPE html><html lang="en-US" class="no-js has-zcm  no-theme is-link-style-exp is-link-order-exp is-link-breadcrumb-exp is-related-search-exp is-vertical-tabs-exp"><head><meta name="description" content="DuckDuckGo. Privacy, Simplified."><meta http-equiv="content-type" content="text/html;
```


however,using  duckduckgo_search with English, worked

<img width="761" alt="image" src="https://github.com/user-attachments/assets/d97d17a4-7bf4-4ff8-b594-d811b266327f">

**Comment by jimmy-walker** [comment]
Thanks your work. @deedy5 
I have met the same problem with Chinese query. It will be fine with English query.
When I open the url in the browser. It remends me of captcha.
Does it mean that my ip is blocked by duckduckgo?

> Try querying in Chinese in your browser. Do you see captcha?

**Comment by yichaosun** [comment]
> Thanks your work. @deedy5 I have met the same problem with Chinese query. It will be fine with English query. When I open the url in the browser. It remends me of captcha. Does it mean that my ip is blocked by duckduckgo?
> 
> > Try querying in Chinese in your browser. Do you see captcha?

用英文行，哪里来的ip屏蔽。。。。

**Comment by chiefluo** [comment]
感谢各位大佬的付出，目前想的折中的办法就是用他自带的翻译将中文翻译成英文，然后回传后再翻译成中文，唯一的弊端是检索到的信息是根据英文检索的。

**Comment by jimmy-walker** [comment]
> > Thanks your work. @deedy5 I have met the same problem with Chinese query. It will be fine with English query. When I open the url in the browser. It remends me of captcha. Does it mean that my ip is blocked by duckduckgo?
> > > Try querying in Chinese in your browser. Do you see captcha?
> 
> 用英文行，哪里来的ip屏蔽。。。。

I  guess the policy of duckduckgo engineering is something like that: It will forbidden the specific ips for chinese query. But it will allow more ip for english query.

**Comment by yichaosun** [comment]
> > > Thanks your work. @deedy5 I have met the same problem with Chinese query. It will be fine with English query. When I open the url in the browser. It remends me of captcha. Does it mean that my ip is blocked by duckduckgo?
> > > > Try querying in Chinese in your browser. Do you see captcha?
> > 
> > 
> > 用英文行，哪里来的ip屏蔽。。。。
> 
> I guess the policy of duckduckgo engineering is something like that: It will forbidden the specific ips for chinese query. But it will allow more ip for english query.

maybe，I have changed to searXNG ，locally

**Comment by chiefluo** [comment]
现在有新的报错了，Failed to invoke tool: Invoke server unavailable error: [tongyi] Server Unavailable Error, <400> InternalError.Algo.InvalidParameter: The messages list should not contain only one message with role "system". Please include at least one other role in the messages.
应该是官方接口的传入参数修改了
![image](https://github.com/user-attachments/assets/642c2c2e-a76f-4a32-93ca-ab9644091e13)

**Comment by deedy5** [comment]
Gentlemen from China, please try v6.3.5 and let me know if it works or not

**Comment by yichaosun** [comment]
> Gentlemen from China, please try v6.3.5 and let me know if it works or not

I will check it as soon as possible,  thx u very much

**Comment by deedy5** [comment]
@yichaosun 

It works for me.
[Try using a proxy](https://github.com/deedy5/duckduckgo_search?tab=readme-ov-file#proxy).

**Comment by cleven1** [comment]
各位大佬们,  有解决方法吗?  求告知

**Comment by deedy5** [comment]
fixed in v6.4.1
